### PR TITLE
Minor: Fix docstring typos

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/arrow-datafusion/pull/7840#pullrequestreview-1686206136

## Rationale for this change

@viirya  found several typos https://github.com/apache/arrow-datafusion/pull/7840#pullrequestreview-1686206136 🤦 

## What changes are included in this PR?
Fix the typos

### CODE REVIEW

**Review by viirya — APPROVED** [review]

File: docs/source/user-guide/configs.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ Environment variables are read during `SessionConfig` initialisation so they mus
...
+| datafusion.execution.parquet.column_index_truncate_length  | NULL                      | Sets column index trucate length                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (alamb) [inline docs/source/user-guide/configs.md:68]: I am not sure why this is required on this PR (maybe the CI didn't fully run on another PR 🤔 ) anyhow it seems a straightforward enough fix

Comment (viirya) [reply docs/source/user-guide/configs.md:68]: Hmm, isn't `truncate` a typo too?

Comment (alamb) [reply docs/source/user-guide/configs.md:68]: 🤔  it is -- but that seems to have been introduced in #7244

https://github.com/apache/arrow-datafusion/pull/7244/files#diff-2f697c310af1a48521829d8bd665cf64b6415fbf88edd370efa30b1ed686354dR328

I am very confused. Let's merge this PR and I'll make another one to fix that typo

Comment (alamb) [reply docs/source/user-guide/configs.md:68]: https://github.com/apache/arrow-datafusion/pull/7874 to fix that one too

**Pull request merged**